metrics/pcp/pcp.go

```diff
@@ -0,0 +1,127 @@
+package pcp
+
+import (
+	"github.com/performancecopilot/speed"
+
+	"github.com/go-kit/kit/metrics"
+)
+
+// Reporter encapsulates a speed client.
+type Reporter struct {
+	c *speed.PCPClient
+}
+
+// NewReporter creates a new Reporter instance.
+// The first parameter is the application name and is used to create the speed client.
+// Hence it should be a valid speed parameter name and should not contain spaces or the path separator
+// for your operating system.
+func NewReporter(appname string) (*Reporter, error) {
+	c, err := speed.NewPCPClient(appname)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Reporter{c}, nil
+}
+
+// Start starts the underlying speed client so it can start reporting registered
+// metrics to your PCP installation.
+func (r *Reporter) Start() { r.c.MustStart() }
+
+// Stop stops the underlying speed client so it can stop reporting registered
+// metrics to your PCP installation.
+func (r *Reporter) Stop() { r.c.MustStop() }
+
+// Counter implements metrics.Counter via a single dimensional speed.Counter.
+type Counter struct {
+	c speed.Counter
+}
+
+// NewCounter creates a new Counter.
+//
+// This requires a name parameter and can optionally take a couple of description
+// strings, that are used to create the underlying speed.Counter and are reported by PCP.
+func (r *Reporter) NewCounter(name string, desc ...string) (*Counter, error) {
+	c, err := speed.NewPCPCounter(0, name, desc...)
+	if err != nil {
+		return nil, err
+	}
+
+	r.c.MustRegister(c)
+	return &Counter{c}, nil
+}
+
+// With is a no-op.
+func (c *Counter) With(labelValues ...string) metrics.Counter { return c }
+
+// Add implements Counter.
+// speed Counters only take int64.
+// speed Gauges can take float64 and Add(float64) is implemented by metrics.Gauge.
```

Comment:
Please fix the capitalization in this and all comment blocks, and clarify the behavior.

```
// Add implements Counter. Speed counters only take int64, so 
// the delta is converted to an int64 before observation.
```